src/main/java/com/exactpro/th2/codec/csv/CsvCodec.java

```diff
@@ -175,12 +175,32 @@ private void decodeCsvData(Collection<ErrorHolder> errors, MessageGroup.Builder
             int headerLength = header.length;
             int rowLength = strings.length;
             for (int i = 0; i < headerLength && i < rowLength; i++) {
-                builder.putFields(header[i], ValueUtils.toValue(strings[i]));
+                int extraLength = getGetExtraLength(i, header, rowLength);
+                if (extraLength == 0) {
+                    builder.putFields(header[i], ValueUtils.toValue(strings[i]));
+                } else {
+                    String[] values = new String[extraLength + 1];
+                    System.arraycopy(strings, i, values, 0, extraLength + 1);
+                    builder.putFields(header[i], ValueUtils.toValue(values));
```

Comment:
I am not sure if we have the overload for arrays in `toValue` function. Can you check it and if we don't have overload - use List instead?

Reply:
> fun Array<*>.toValue(): Value = iterator().toValue()

we have it